apps/mobile/src/scripts/injected-provider.ts

```diff
@@ -0,0 +1,67 @@
+export default `
+  function createEventListeners() {
+    let eventListeners = {};
+    return {
+      addEventListener(type, handler) {
+        eventListeners[type] = handler;
+      },
+      dispatchEvent(type, data) {
+        if (eventListeners[type]) {
+          eventListeners[type](data);
+        }
+      },
+      removeEventListener(type) {
+        eventListeners[type] = undefined;
+      },
+    };
+  }
+
+  const eventListeners = createEventListeners();
+
+  const provider = {
+    isLeather: true,
+
+    getURL() {
+      // window.ReactNativeWebView.postMessage('getURL');
+    },
+    async request(method, params) {
+      const id = crypto.randomUUID();
+      const rpcRequest = {
+        jsonrpc: '2.0',
+        id,
+        method,
+        params,
+      };
+      window.ReactNativeWebView.postMessage(JSON.stringify(rpcRequest));
```

Comment:
Great that this doesn't look too complicated. I wonder if there's any more generic way we can great the `inpage.ts` script in the extension, such that we could reuse it here.